File: minFactorDist.js
function minDistance(n){
    let factors = []
    let minDist = n
  
    for (let i = 1; i <= n; i++) {
      if (n % i === 0) {
        factors.push(i)
      }
    }
  
    for (let i = 0; i < factors.length - 1; i++) {
      if (factors[i + 1] - factors[i] < minDist) {
        minDist = factors[i + 1] - factors[i]
      }
    }
    return minDist
  }


console.log(minDistance(8))
console.log(minDistance(25))
console.log(minDistance(13013))

File: min_factor_dist.py
def shortest_intance(n):
    factors = []
    shortest_int = n
    for num in range(1,(n+1)):
        if n % num == 0:
            factors.append(num)
    for i in range(len(factors)-1):
        if factors[i+1] - factors[i] < shortest_int:
            shortest_int = factors[i+1] - factors[i]
    return(shortest_int)


print(shortest_intance(8))
print(shortest_intance(25))
print(shortest_intance(13013))